operator/v1/types_ingress.go

```diff
@@ -282,6 +287,339 @@ var (
+	// to use DES-CBC3-SHA but not EDH-DSS-DES-CBC3-SHA (yaml):
+	//
+	//   ciphers:
+	//     - DES-CBC3-SHA
+	//     - !EDH-DSS-DES-CBC3-SHA
+	//
+	// If unset, Ciphersuites are determined by the SecurityProfileType.
+	//
+	// +optional
+	Ciphers []*string `json:"ciphers,omitempty"`
+	// securityProtocol is used to specify one or more encryption protocols
+	// that are negotiated during the SSL/TLS handshake with the IngressController.
+	//
+	// If unset, the security protocol is determined by the SecurityProfileType.
+	//
+	// +optional
+	SecurityProtocol *SecurityProtocol `json:"securityProtocol,omitempty"`
+	// dhParamSize sets the maximum size of the Diffie-Hellman parameters used for generating
+	// the ephemeral/temporary Diffie-Hellman key in case of DHE key exchange. The final size
+	// will try to match the size of the server's RSA (or DSA) key (e.g, a 2048 bits temporary
+	// DH key for a 2048 bits RSA key), but will not exceed this maximum value.
+	//
+	// If unset, the DH parameter size is determined by the SecurityProfileType.
+	//
+	//   SecurityProfileType Modern:       DHParamSize2048
+	//   SecurityProfileType Intermediate: DHParamSize2048
+	//   SecurityProfileType Old:          DHParamSize1024
+	//
+	// Supported DH Parameter sizes are:
+	//
+	//   "2048": A Diffie-Hellman parameter of 2048 bits.
+	//   "1024": A Diffie-Hellman parameter of 1024 bits.
+	//
+	// +optional
+	DHParamSize *DHParamSize `json:"dhParamSize,omitempty"`
+}
+
+// SecurityProtocol defines one or more security protocols used by
+// an IngressController to secure network connections.
+type SecurityProtocol struct {
+	// minimumVersion enforces use of SecurityProtocolVersion or newer on
+	// SSL connections initiated by an IngressController. minimumVersion
+	// must be lower than or equal to maximumVersion.
+	//
+	// If unset and maximumVersion is set, minimumVersion will be set
+	// to maximumVersion. If minimumVersion and maximumVersion are unset,
+	// the minimum version is determined by the SecurityProfileType.
+	//
+	//   SecurityProfileType Modern:       SecurityProtocolTLS12Version
+	//   SecurityProfileType Intermediate: SecurityProtocolTLS10Version
+	//   SecurityProfileType Old:          SecurityProtocolTLS10Version
+	//
+	// Supported minimum versions are:
+	//
+	//   "TLSv1.3": Version 1.3 of the TLS security protocol used for securing
+	//   IngressController network connections.
+	//   "TLSv1.2": Version 1.2 of the TLS security protocol used for securing
+	//   IngressController network connections.
+	//   "TLSv1.1": Version 1.1 of the TLS security protocol used for securing
+	//   IngressController network connections.
+	//   "TLSv1.0": Version 1.0 of the TLS security protocol used for securing
+	//   IngressController network connections.
+	//
+	// +optional
+	MinimumVersion *SecurityProtocolVersion `json:"minimumVersion,omitempty"`
+	// maximumVersion enforces use of SecurityProtocolVersion or older on
```

Comment:
The wording here is strange.  Maybe it is missing a couple words: "maximumVersion enforces use of _the specified_ SecurityProtocolVersion or older"?